### PR TITLE
Re-enable sidecar_api_test

### PR DESCRIPTION
**Pull request by howardjohn**
Changes were required because:
* mgmtCluster was changed - I guess it used to be 2 ports 3333 and 9999, but now we create it as 15020 by default, but in the tests we never set the management ports. I am not sure if this should be removed, or if we should find a way to trigger a management port being created. I dug through the code a bit but wasn't sure how to do this. @costinm or @rshriram any ideas? I don't want to disable the test without cause
* BlackHoleCluster -> PassthroughCluster because ALLOW_ANY is default now

### CODE REVIEW

File: tests/integration/pilot/sidecar_api_test.go [hunk 1]
```diff
@@ -169,5 +149,5 @@ func validateMongoListener(t *testing.T, response *structpath.Instance) {
 // - Do cleanup before exit
 // - process testing specific flags
 func TestMain(m *testing.M) {
-	// framework.Main("sidecar_api_test", m, istio.SetupOnKube(&ist, nil))
+	framework.Main("sidecar_api_test", m, istio.SetupOnKube(&ist, nil))
```
Comment (ozevren) [inline tests/integration/pilot/sidecar_api_test.go]: You can remove istio.SetupOnKube part for the time being. The tests are already tagged as native only at this point.

**Comment by ozevren** [comment]
Integration tests are still failing though. :)

**Comment by howardjohn** [comment]
> Integration tests are still failing though. :)

I think the tests are flakey, so we shouldnt enable them until that is fixed. We are getting this error:
```
2019-03-27T23:19:53.693736Z     info    transport: loopyWriter.run returning. connection error: desc = "transport is closing"
2019-03-27T23:19:53.693766Z     info    transport: loopyWriter.run returning. connection error: desc = "transport is closing"
2019-03-27T23:19:53.693774Z     info    ads     ADS: "[::1]:46296" sidecar~10.2.0.1~app3.testns~testns.cluster.local-1 terminated EOF
2019-03-27T23:19:53.693813Z     info    pickfirstBalancer: HandleSubConnStateChange: 0xc00048cf40, TRANSIENT_FAILURE
2019-03-27T23:19:53.693844Z     info    pickfirstBalancer: HandleSubConnStateChange: 0xc00048cf40, CONNECTING
2019-03-27T23:19:53.694125Z     info    grpc: addrConn.createTransport failed to connect to {[::]:36571 0  <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp [::]:36571: connect: connection refused". Reconnecting...
2019-03-27T23:19:53.694181Z     info    pickfirstBalancer: HandleSubConnStateChange: 0xc00048cf40, TRANSIENT_FAILURE
2019-03-27T23:19:54.694049Z     info    pickfirstBalancer: HandleSubConnStateChange: 0xc00048cf40, CONNECTING
2019-03-27T23:19:54.694361Z     info    grpc: addrConn.createTransport failed to connect to {[::]:36571 0  <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp [::]:36571: connect: connection refused". Reconnecting...
```

This repeats forever

**Pull request merged**